wire.md

```diff
@@ -0,0 +1,1008 @@
+`text`         | `u8[text_len] `    | chat message text (UTF-8)
+
+`post_type` MUST be set to `0`.
+
+The `text` body of a chat message MUST be a valid UTF-8 string. Its length MUST
+NOT exceed 4 kibibytes (4096 bytes).
+
+#### 6.2.3 `post/delete`
+
+Request that peers encountering this post delete the referenced posts from
+their local storage, and not store the referenced posts in the future.
+
+field           | type                   | desc
+----------------|------------------------|-------------------------
+`num_deletions` | `varint`               | how many hashes of posts there are to be deleted
+`hashes`        | `u8[32*num_deletions]` | concatenated hashes of posts to be deleted
+
+`post_type` MUST be set to `1`.
+
+A client interpreting this post MUST only perform a local deletion of the
+referenced posts if the author (`post.public_key`) matches the author of the
+post to be deleted (i.e. only the user who authored a post may delete it).
+
+`num_deletions` MUST be set to 1 or greater.
+
+#### 6.2.4 `post/info`
+
+Set public information about one's self.
+
+field        | type               | desc
+-------------|--------------------|-------------------------
+`key1_len`   | `varint`           | length of the first key to set, in bytes
+`key1`       | `u8[key1_len]`     | name of the first key to set (UTF-8)
+`value1_len` | `varint`           | length of the first value to set, belonging to `key1`, in bytes
+`value1`     | `u8[value_len]`    | value of the first key:value pair
+...          |                    |
+`keyN_len`   | `varint`           | length of the Nth key to set, in bytes
+`keyN`       | `u8[keyN_len]`     | name of the Nth key to set (UTF-8)
+`valueN_len` | `varint`           | length of the Nth value to set, belonging to `keyN`, in bytes
+`valueN`     | `u8[value_len]`    | value of the Nth key:value pair
+
+`post_type` MUST be set to `2`.
+
+Several key/value pairs MAY be set at once. A post MUST indicate it is done
+specifying pairs by setting the final `keyN_len` to 0.
+
+A `post/info` post is a complete description of a user's self-published
+information. A `post/info` post fully replaces a previously known version, and
+are not additive. If a key is set in one `post/info` but not the subsequent
+one, that key MUST be treated as being set to its default. See the table of
+keys/values below for each key's default value.
+
+Keys MUST be UTF-8 strings, and MUST be between 1 and 128 codepoints in length.
+
+A value field MUST NOT exceed 4096 bytes (4 kibibytes) in length.
+
+The valid bytes for a value depends on the key. See the table below.
+
+The following keys SHOULD be supported:
+
+key       | value format | desc
+----------|--------------|---------------------------------------
+`name`    | UTF-8        | the name this user wishes to be known as. Default value is "" (empty string).
+
+To save space, a client MAY discard older versions of these messages for a
+particular user.
```

Comment:
This makes it sound like there are other cases where clients are forbidden to get rid of some data. Which feels odd.

Reply:
Oh interesting. I had actually the opposite intention: to tell implementors that they're under no obligation to hold onto things forever.

Reply:
Right. I read it as "for these particular things, you MAY forget them, but everything where we don't give you express permission has to stay". Perhaps something along "Clients MAY discard any data they please." somewhere in a prominent position?

Reply:
Great idea. This is now live, under "5.1.5 Keeping & Discarding Posts".